[MaxLeong] iP

Mmaxx15 opened the pull request.

Mmaxx15 committed:
Rename Duke to Dobby

Mmaxx15 committed:
Rename Duke to Dobby

Mmaxx15 committed:
Rename Duke to Dobby

Mmaxx15 committed:
Rename Duke to Dobby

Mmaxx15 committed:
spacing

Mmaxx15 committed:
fixed indentation

Mmaxx15 committed:
added Echo function to Dobby

Mmaxx15 committed:
Add ability to add tasks

Mmaxx15 committed:
Added ability to list tasks

Mmaxx15 committed:
Added ability to mark tasks

Mmaxx15 committed:
modified markTask and unmarkTask

Mmaxx15 committed:
edited to fit coding standards and gave bby more character

Mmaxx15 committed:
refactored task.java into task.java and taskList.java

Mmaxx15 committed:
Created Deadline and Events classes for Level-4

Mmaxx15 committed:
moved functions related to marking and unmarking tasks into new class called Todo.java

Mmaxx15 committed:
Revert "moved functions related to marking and unmarking tasks into new class called Todo.java"

This reverts commit 41feeeb284787f778310638f58f937007d841820.

Mmaxx15 committed:
reformatted again and added deadline class

eliztan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,45 @@
+        return "[" + getStatusIcon() + "] " + description;
+    }
+
+    /**
```
Good job doing documentation for all your methods

jthibault03 commented on src/main/java/Dobby.java:
I like the dobby word art

eliztan commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+         * @param taskIndex index of task stored in array
+         */
+        public void unmarkTask(int taskIndex) {
+            if (!isValidIndex(taskIndex)) {
```
You should considering following the "happy path" as much as possible

jthibault03 commented on src/main/java/TaskList.java:
good coding standard stuff with all caps for max tasks

jthibault03 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+import java.util.Scanner;
+public class TaskList {
+        private final Task[] taskList;
```
indent should only be 4 spaces i think, and is a bit inconsistent in this file. maybe fix a bit to match coding standard. but i like all the cases to ensure the user is entering input in the right format – very thorough of you!

eliztan commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,12 @@
+//seems useless can probably just delete
```
You should considering deleting it if you're not using the class

jthibault03 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,45 @@
+public class Task {
+    protected String description;
+    protected Boolean isDone;
```
the word boolean should be lowercase since it's a primitive data type?

jthibault03 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+                command = in.nextLine().toLowerCase();
+                String[] commandParts = command.split(" ", 2);
+
+                switch (commandParts[0]) {
```
good use of switch commands in general!

eliztan commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+         */
+        public void addTask(Task task) {
+            taskList[taskCount] = task;
+            //isMarked[taskCount] = false;
```
you should consider deleting out commented code to not confuse the person reading it

Mmaxx15 committed:
Completed Events

Mmaxx15 committed:
added ability to check if unmark index is invalid

Mmaxx15 committed:
fixed indentation

Mmaxx15 committed:
Coding standards for "Events" invalid command

Mmaxx15 committed:
Manage exception for invalid input when marking and unmarking tasks

Mmaxx15 committed:
extracted mark and unmark task as methods for readability

Mmaxx15 committed:
extracted addDeadlineTask and addEvent as methods for readability

Mmaxx15 committed:
renamed markTask and unmarkTask due to duplicate names

Mmaxx15 committed:
handle exception when tasklist is full

Mmaxx15 committed:
removed todo class

LouisLouis19 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,158 @@
+public class TaskList {
+        private final Task[] taskList;
+        private static final int MAX_TASKS = 100;
+        //protected boolean[] isMarked;
```
You could consider removing unused comment if it is no longer needed

LouisLouis19 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,158 @@
+                System.out.println("Invalid deadline format! Use: deadline <<description>> /by <<deadline>>.");
+            } else {
+                String[] endDateParts = eventparts[1].split("/to" , 2);
+                if (endDateParts.length != 2) {
```
You could consider refactoring this nested conditional to prevent deep nesting and adhere to this code quality [guideline](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting)

LouisLouis19 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,158 @@
+            this.taskCount = 0;
+        }
+
+    private void processTaskCommand(String command) {
```
You could consider refactoring parts of this method to separate methods to avoid long methods and adhere to this code quality [guideline](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting)

LouisLouis19 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,158 @@
+            for (int i = 0 ; i < taskCount ; i += 1) {
+                System.out.println("  " + (i+1) + ". [" + taskList[i].getType() + "]" + taskList[i]);
+            }
+            System.out.println("~~~~~~~~~~~~~~~~");
```
In general, you could consider avoiding some reused string literals (not restricted to this method and file) by refactoring to a named constant to adhere to this code quality [guideline](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers)

Mmaxx15 committed:
Merge branch 'branch-Level-5'

Mmaxx15 committed:
Implemented ArrayList for storing tasks

Mmaxx15 committed:
added function to delete tasks and changed fixed array to dynamic array

Mmaxx15 committed:
Added method to save tasks into a text file which updates everytime a task is added,marked or unmarked

Mmaxx15 committed:
Added ability to load file from data/text.txt
description of events and deadline are in format of user inputs

however current iteration prints "add <task>" when running of dobby as methods addTask/addEvent/addDeadlineTasks from class TaskList is used

Mmaxx15 committed:
modifed methods addTask,addEvent and addDeadlineTasks by adding additional boolean input which checks if the method is called from user input of when loading file from text.txt

Mmaxx15 committed:
removed spacing

Mmaxx15 committed:
mergerged branch-level-6 to master and merged branch-Level-7 to master
managed conflicts during merger

delete method does not append text.txt

Mmaxx15 committed:
Delete now modifies changed the text.txt file

Mmaxx15 committed:
fixed typo in dobby name

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
made Ui class to display text user sees

Mmaxx15 committed:
removeed commented code

Mmaxx15 committed:
Un-hardcoded the file path to make it usable on other devices

Mmaxx15 committed:
extracted repeat print lines for codequality

Mmaxx15 committed:
extracted out ProcessCommands class to handle user commands, leaving TaskList class to deal with task related operations

Mmaxx15 committed:
??

Mmaxx15 committed:
added method to find tasks

Mmaxx15 committed:
Added method comments

Mmaxx15 committed:
added comments for methods in ProcessCommands class

Mmaxx15 committed:
Merge pull request #1 from Mmaxx15/branch-Level-9

Branch level 9

Mmaxx15 committed:
Merge pull request #2 from Mmaxx15/A-JavaDoc

A-JavaDoc

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Unhardcoded the file path so it works on any computer with java 11

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
resolved error of cant find path

Mmaxx15 committed:
editted comments

Mmaxx15 committed:
extracted methods from loadtasksfromfile to reduce arrow-head

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
comments for methods in filemanager class

Mmaxx15 committed:
give pronous in comment. OK Intellij

Mmaxx15 committed:
Revert "give pronous in comment. OK Intellij"

This reverts commit ab6485a4c644a14109aa97b25230ab18d467b98f.

Mmaxx15 committed:
Revert "Revert "give pronous in comment. OK Intellij""

This reverts commit cefcff548f0def7b73b568346f9d016fd2fd9bce.

Mmaxx15 committed:
tidy up code

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
Update README.md

Mmaxx15 committed:
removed unnecessary comments and deleted empty todo class

Mmaxx15 committed:
removed throwing exception and replaced with print lines
